src/hooks/useSupplyWithdrawModal/Modal/Supply/form.tsx

```diff
@@ -0,0 +1,281 @@
+        />
+      )}
+
+      <FormikTokenTextField
+        data-testid={TEST_IDS.valueInput}
+        name="amount"
+        token={asset.vToken.underlyingToken}
+        disabled={isTransactionLoading || isSupplyingLuna}
+        rightMaxButton={{
+          label: t('supplyWithdraw.max').toUpperCase(),
+          valueOnClick: maxInput.toFixed(),
+        }}
+        css={styles.input}
+        // Only display error state if amount is higher than borrow limit
+        displayableErrorCodes={[ErrorCode.HIGHER_THAN_MAX]}
+      />
+
+      <Typography
+        component="div"
+        variant="small2"
+        css={[styles.greyLabel, styles.getRow({ isLast: true })]}
+      >
+        <Trans
+          i18nKey={inputLabel}
+          components={{
+            White: <span css={styles.whiteLabel} />,
+          }}
+          values={{
+            amount: formatTokensToReadableValue({
+              value: maxInput,
+              token: asset.vToken.underlyingToken,
+            }),
+          }}
+        />
+      </Typography>
+
+      {tokenInfo.map((info, index) => (
+        <LabeledInlineContent
+          css={styles.getRow({ isLast: index === tokenInfo.length - 1 })}
+          className="info-row"
+          {...info}
+          key={info.label}
+        />
+      ))}
+
+      <Delimiter css={styles.getRow({ isLast: true })} />
+
+      <BorrowBalanceAccountHealth
+        css={styles.getRow({ isLast: true })}
+        borrowBalanceCents={userTotalBorrowBalanceCents.toNumber()}
+        borrowLimitCents={
+          hypotheticalBorrowLimitCents?.toNumber() || userTotalBorrowLimitCents.toNumber()
+        }
+        safeBorrowLimitPercentage={SAFE_BORROW_LIMIT_PERCENTAGE}
+      />
+
+      <LabeledInlineContent
+        label={t('supplyWithdraw.supplyBalance', {
+          tokenSymbol: asset.vToken.underlyingToken.symbol,
+        })}
+        css={styles.getRow({ isLast: false })}
+        className="info-row"
+      >
+        <ValueUpdate
+          original={asset.userSupplyBalanceTokens}
+          update={hypotheticalTokenSupplyBalance}
+          format={(value: BigNumber | undefined) =>
+            formatTokensToReadableValue({
+              value,
+              token: asset.vToken.underlyingToken,
+              minimizeDecimals: true,
+              addSymbol: false,
+            })
+          }
+        />
+      </LabeledInlineContent>
+
+      <LabeledInlineContent
+        label={t('supplyWithdraw.borrowLimit')}
+        css={styles.getRow({ isLast: false })}
+        className="info-row"
+      >
+        <ValueUpdate original={userTotalBorrowLimitCents} update={hypotheticalBorrowLimitCents} />
+      </LabeledInlineContent>
+
+      <LabeledInlineContent
+        label={t('supplyWithdraw.dailyEarnings')}
+        css={styles.getRow({ isLast: true })}
+        className="info-row"
+      >
+        <ValueUpdate original={dailyEarningsCents} update={hypotheticalDailyEarningCents} />
+      </LabeledInlineContent>
```

Comment:
I'd put this in a separate `AccountData` component like for the borrow modal, so it can be shared between the Supply and Withdraw tabs